### PR TITLE
Add display policy url

### LINKED ISSUE #36: Add URL fields for the different policies + Add popup to view URLs to various policies owned by a client

### PR DESCRIPTION
**Pull request by jlxw48**
Did not manage to create a deep link to the browser; still not sure if it is possible. Will check in the next iteration.

Here, policies are given optional URLs. Tests are also fixed to include testing for these URLs, if any. Also, as per the title, a popup window showing the policies associated with the selected contact was implemented.

### CODE REVIEW

File: src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json [hunk 1]
```diff
@@ -6,43 +6,48 @@
     "email" : "alice@example.com",
     "address" : "123, Jurong West Ave 6, #08-111",
     "tagged" : [ "friends" ],
-    "policies" : ["#P12345"]
+    "policies" : ["Policy_1111>null"]
```
Comment (NBH99) [inline src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json:9]: The string "null" could be "no_url" as the insurance agent may not know what is a null when checking the json file.

File: src/main/java/seedu/address/storage/JsonAdaptedInsurancePolicy.java [hunk 2]
```diff
@@ -8,35 +8,63 @@
...
+        if (policyIdAndUrl.length == 1) {
+            return new String[] {policyIdAndUrl[0], null};
+        }
+
+        policyIdAndUrl[1] = definePolicyUrl(policyIdAndUrl[1]);
+        return policyIdAndUrl;
+    }
+
+    private static String definePolicyUrl(String urlIfPresent) {
+        if (urlIfPresent.equals("null")) {
```
Comment (NBH99) [inline src/main/java/seedu/address/storage/JsonAdaptedInsurancePolicy.java:52]: The string "null" could be "no_url" as the insurance agent may not know what is a null when checking the json file.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -125,18 +131,46 @@ public static Tag parseTag(String tag) throws ParseException {
...
+     * @throws ParseException if the given {@code policy} is invalid.
+     */
+    public static InsurancePolicy parsePolicy(String policy) throws ParseException {
+        requireNonNull(policy);
+        String trimmedPolicy = policy.trim();
+
+        if (!InsurancePolicy.isValidPolicyInput(trimmedPolicy)) {
+            throw new ParseException(InsurancePolicy.MESSAGE_CONSTRAINTS);
+        }
+        logger.info("HIIIII" + trimmedPolicy);
```
Comment (NBH99) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:147]: The log message could be more appropriate.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -125,18 +131,46 @@ public static Tag parseTag(String tag) throws ParseException {
...
+        if (InsurancePolicy.isPolicyId(trimmedPolicy)) {
+            return new InsurancePolicy(trimmedPolicy);
+        }
+
+        // Else contains URL too
+        String[] idAndUrl = trimmedPolicy.split(">", 2);
+        String policyId = idAndUrl[0];
+        String policyUrl = idAndUrl[1];
+
+        logger.info("BYEEEE" + Arrays.toString(idAndUrl));
```
Comment (NBH99) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:159]: The log message could be more appropriate.

**Review by NBH99 — COMMENTED** [review]
Look good to me, do remember to create some test for testing.

File: src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json (hunk 1 above)
Comment (jlxw48) [reply src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json:9]: Great suggestion! Thanks!

**Pull request closed without merging**